src/transformers/adapters/peft_mixin.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from typing import Optional
+
+from ..utils import ADAPTER_CONFIG_NAME, cached_file, logging, requires_backends
+
+
+logger = logging.get_logger(__name__)
+
+
+class PeftAdapterMixin:
+    """
+    A class containing all functions for loading and using adapters weights that are supported in PEFT library.
+    Currently supported PEFT methods are all non-prefix tuning methods
+    """
+
+    def load_adapter(
+        self,
+        peft_model_id: str,
+        adapter_name: Optional[str] = "default",
+        revision: Optional[str] = None,
+        use_auth_token: Optional[str] = None,
+        commit_hash: Optional[str] = None,
+    ):
+        """
+        Load adapter weights from file. Requires peft as a backend to load the adapter weights
+        """
+        requires_backends(self.load_adapter, "peft")
```

Comment:
Technically, this also requires a specific peft version, right? But it seems that requiring a version is not supported.

Reply:
Yes I can add that in the next line

Reply:
Agree it would be nice to get a very good error message here